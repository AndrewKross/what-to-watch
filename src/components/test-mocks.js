export const PromoFilmData = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const GENRE = `drama`;
const RELEASE = `2020`;
const RATING = 9;
const VOTES = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://cdn.videvo.net/videvo_files/video/free/2019-09/small_watermarked/190828_07_MarinaBayatNightDrone_UHD_02_preview.webm`;
const RUN_TIME = 140;

const DESCRRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`
];

const review = {
  id: `0`,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date()
};

export const films = TITLES.map((title, index) => {
  return ({
    id: index.toString(),
    title,
    cover: COVERS[index],
    poster: COVERS[index],
    preview: PREVIEW,
    genre: GENRE,
    release: RELEASE,
    rating: RATING,
    votes: VOTES,
    description: DESCRRIPTION,
    director: DIRECTOR,
    actors: ACTORS,
    runTime: RUN_TIME,
    reviews: [review]
  });
});

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`
];
