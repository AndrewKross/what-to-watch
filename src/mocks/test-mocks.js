import { NameSpace } from '../reducer/reducer';
import { DEFAULT_GENRE, LoadingStatus } from '../const';

const RELEASE = 2020;
const RATING = 9;
const RATINGS_COUNT = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
const RUN_TIME = 140;
const COMMENTS_COUNT = 6;

const DESCRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`,
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`,
];

export const GENRES = [
  `all genres`,
  `comedy`,
  `drama`,
  `melodrama`,
  `horror`,
  `crime`,
  `documentary`,
  `romance`,
  `thriller`,
];

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`,
];

const comment = {
  id: 0,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date(`December 25, 2019 01:00:00`),
};

export const comments = Array(COMMENTS_COUNT)
  .fill(``)
  .map(() => comment);

export const films = TITLES.map((title, index) => {
  return ({
    id: index,
    title,
    cover: COVERS[index],
    posterImage: COVERS[index],
    previewImage: COVERS[index],
    backgroundImage: COVERS[index],
    previewVideo: PREVIEW,
    videoMain: PREVIEW,
    genre: GENRES[index + 1] || GENRES[1],
    released: RELEASE,
    rating: RATING,
    ratingsCount: RATINGS_COUNT,
    description: DESCRIPTION,
    director: DIRECTOR,
    starring: ACTORS,
    runTime: RUN_TIME,
    isFavorite: false,
    backgroundColor: `black`,
  });
});

export const userData = {
  id: 1,
  email: `Andrew@gmail.com`,
  name: `Andrew`,
  avatar: `img/1.png`,
};

export const loginData = {
  email: `Andrew@gmail.com`,
  password: `1234`,
};

export const reviewToPost = {
  rating: 8,
  comment: `Great!`,
};

export const defaultMockStore = {
  [NameSpace.DATA]: {
    films,
    comments,
    isFilmsLoaded: true,
    isCommentsLoaded: true,
  },
  [NameSpace.APP_STATE]: {
    filmsOnScreen: 8,
    currentGenre: DEFAULT_GENRE,
  },
  [NameSpace.USER]: {
    isAuthorized: true,
    userInfo: userData,
    reviewLoadingStatus: LoadingStatus.PENDING,
    authorizationLoadingStatus: LoadingStatus.PENDING,
  },
};
