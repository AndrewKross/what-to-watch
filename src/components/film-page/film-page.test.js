import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import FilmPage from "./film-page.jsx";
import { mockStore } from "../../test-mocks";

test(`FilmPage should render correctly`, () => {
  const mockStoreConfig = configureStore([]);
  const store = mockStoreConfig(mockStore);
  const tree = renderer
    .create(
        <Provider store={store}>
          <FilmPage
            selectedFilm={mockStore.filteredFilms[0]}
            films={mockStore.filteredFilms}
            onFilmCardClick={() => { }}
          />
        </Provider>, {
          createNodeMock: () => ({}),
        },
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
