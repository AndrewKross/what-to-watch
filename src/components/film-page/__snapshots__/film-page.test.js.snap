// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should filmPage render correctly 1`] = `
<filmPage
  filmsList={
    Array [
      Object {
        "actors": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
        ],
        "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
        "director": "Wes Andreson",
        "genre": "Action",
        "id": "film-1",
        "poster": "./img/revenant.jpg",
        "preview": "https://download.blender.org/durian/preview/sintel_preview-480p.mp4",
        "rating": "5,6",
        "release": "2017",
        "runTime": "1h 35m",
        "title": "Revenant",
        "ratingsCountCount": "278",
      },
    ]
  }
  onFilmCardClick={[Function]}
  selectedFilm={
    Object {
      "actors": Array [
        "Bill Murray",
        "Edward Norton",
        "Jude Law",
        "Willem Dafoe",
      ],
      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "director": "Wes Andreson",
      "genre": "Action",
      "id": "film-1",
      "poster": "./img/revenant.jpg",
      "preview": "https://download.blender.org/durian/preview/sintel_preview-480p.mp4",
      "rating": "5,6",
      "release": "2017",
      "runTime": "1h 35m",
      "title": "Revenant",
      "ratingsCountCount": "278",
    }
  }
/>
`;
